scripts/data/client.py

```diff
@@ -16,32 +15,51 @@
 from format_args import format_args
 import logging
 from logging.handlers import TimedRotatingFileHandler
+import signal
+import sys
 
 logger = logging.getLogger(__name__)
 
 # Constants
 MAX_WEIGHT_LIMIT = 8000  # Total weight limit for all jobs
 THREAD_POOL_SIZE = os.cpu_count()  # Number of threads for processing
 QUEUE_MAX_SIZE = THREAD_POOL_SIZE * 2  # Maximum size of the job queue
-
 BASE_DIR = Path(".client_cache")
 
 # Shared state variables
 current_weight = 0
 weight_lock = threading.Condition()
 job_queue = queue.Queue(maxsize=QUEUE_MAX_SIZE)
 
+shutdown_event = threading.Event()
+executor = None  # Global executor reference for shutdown
+
+
+def handle_sigterm(signum, frame):
+    """Handle SIGTERM by initiating immediate shutdown"""
+    logger.info("Received SIGTERM signal. Initiating immediate shutdown...")
+    shutdown_event.set()
+
+    # Clear the job queue
+    while not job_queue.empty():
```

Comment:
signal handler modifying `job_queue` does not look like a good design. I would just set `shutdown_event` here. 
In `job_producer` when shutdown is detected all consumers can be notified with `notify_all`. Cleaning `job_queue` does not seem necessary.